src/cargo-fmt/main.rs

```diff
@@ -293,11 +294,19 @@ fn get_targets_recursive(
                 continue;
             }
 
-            let mut manifest_path = PathBuf::from(&package.manifest_path);
-
-            manifest_path.pop();
-            manifest_path.push(&dependency.name);
-            manifest_path.push("Cargo.toml");
+            let dependency_package = metadata_with_deps
+                .packages
+                .iter()
+                .find(|p| p.name == dependency.name);
+            let manifest_path = if dependency_package.is_some() {
+                PathBuf::from(&dependency_package.unwrap().manifest_path)
+            } else {
```

Comment:
I'm not sure if there's a circumstance where the dependency name wouldn't be found/when the `if` block woudln't be executed, but have this `else` block to fallback to the previous behavior just in case.